src/main/java/com/kneelawk/extramodintegrations/util/stack/EntityEmiStack.java

```diff
@@ -21,22 +22,33 @@
 
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 public class EntityEmiStack extends EmiStack {
     private static final Map<EntityType<?>, Entity> CACHE = new HashMap<>();
+    private static final Set<EntityType<?>> BLOCKLIST = new HashSet<>();
 
-    private final Entity entity;
     private final EntityType<?> type;
 
     public EntityEmiStack(EntityType<?> entityType) {
         this.type = entityType;
         MinecraftClient client = MinecraftClient.getInstance();
-        this.entity = CACHE.computeIfAbsent(entityType, entityType1 -> {
-            if (entityType1 == EntityType.PLAYER) return client.player;
-            return entityType.create(client.world);
-        });
+        if (!BLOCKLIST.contains(entityType)) {
+            CACHE.computeIfAbsent(entityType, entityType1 -> {
+                if (entityType1 == EntityType.PLAYER) return client.player;
+                try {
+                    return entityType.create(client.world);
+                } catch (Throwable t) {
+                    BLOCKLIST.add(entityType1);
+                    ExMIMod.LOGGER.error("Failed to construct entity");
+                    t.printStackTrace();
```

Comment:
This is a bad idea. The error will be lost in `latest.log`, and might not show up with all launchers. Instead you should pass the throwable as the last parameter of the `Logger#error` call, which adds it to the log message.

Reply:
Correction: the error will show up, but as an ugly `STDOUT` message

Reply:
done